src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -310,6 +311,17 @@ public boolean isProperty(String name) {
         return options.javafx() && name.endsWith("Property");
     }
 
+    /*
+     * If a similar query is added to javax.lang.model, use that instead.
+     */
+    public static boolean isImplicitlyDeclaredClass(Tree e) {
+        if (e.getKind() != Tree.Kind.CLASS)
+            return false;
+        if (!(e instanceof com.sun.tools.javac.tree.JCTree.JCClassDecl classDecl))
+            return false; // this should probably never happen with jdk.compiler
+        return (classDecl.mods.flags & com.sun.tools.javac.code.Flags.IMPLICIT_CLASS) != 0;
+    }
+
```

Comment:
Generally, this should all be in the `Workarounds` class, which is the intended enclosure for any code that needs to access `javac` internals that is not directly available via any public API.

Alternatively, you could put this now into `Trees` (not `java.lang.model`, since this is an AST-only method, not an `Element`-based method.